e3nn/rs.py

```diff
@@ -271,14 +271,45 @@ def sorted_truncated_tensor_product(Rs_1, Rs_2, lmax):
 
     example: sorted_truncated_tensor_product([(1, 1), (2, 2)], [(2, 2)], 1) = ([(4, 0, 0), (6, 1, 0)], matrix)
     """
-    Rs, Q = tensor_product_in_in(Rs_1, Rs_2, partial(o3.selection_rule, lmax=lmax))
+    Rs, Q = tensor_product(Rs_1, Rs_2, partial(o3.selection_rule, lmax=lmax))
     Rs, perm = sort(Rs)
     Rs = simplify(Rs)
     matrix = torch.einsum('ij,jkl->ikl', perm, Q)
     return Rs, matrix
 
 
-def tensor_product_in_in(Rs_in1, Rs_in2, get_l_output=o3.selection_rule, normalization='component'):
+def tensor_product(input1, input2, output, normalization='component'):
+    """
+    Compute the matrix Q
+    from Rs_out to Rs_in1 tensor product with Rs_in2
+
+    For normalization='component',
+    The set of "lines" { Q[i] }_i is orthonormal
+
+    :return: Rs_missing, Q
+
+    examples:
+    Rs_out, Q = tensor_product_in_in(Rs_in1, Rs_in2, selection_rule)
+    # Rs_in1 x Rs_in2 -> Rs_out
+    torch.einsum('kij,i,j->k', Q, A, B)
+
+    Rs_in2, Q = tensor_product_in_in(Rs_in1, selection_rule, Rs_out)
+    # Rs_in1 x Rs_in2 -> Rs_out
+    torch.einsum('kij,i,j->k', Q, A, B)
+    """
+    if isinstance(input1, list) and isinstance(input2, list):
+        return _tensor_product_in_in(input1, input2, output, normalization)
+
+    if isinstance(input1, list) and isinstance(output, list):
+        return _tensor_product_in_out(input1, input2, output, normalization)
+
+    if isinstance(input2, list) and isinstance(output, list):
+        Rs_in1, Q = _tensor_product_in_out(input2, input1, output, normalization)
+        Q = torch.einsum('kij->kji', Q)
+        return Rs_in1, Q
```

Comment:
Sofar it's just like that but later I will look if the functions can be really merged into one